src/DurableTask.Netherite/StorageLayer/Faster/LogWorker.cs

```diff
@@ -123,12 +126,30 @@ void AddToFasterLog(byte[] bytes)
                     bool isLastFragment = 1 + bytes.Length - pos <= this.maxFragmentSize;
                     int packetSize = isLastFragment ? 1 + bytes.Length - pos : this.maxFragmentSize;
                     bytes[pos - 1] = (byte)(((pos == 1) ? first : none) | (isLastFragment ? last : none));
-                    this.log.Enqueue(new ReadOnlySpan<byte>(bytes, pos - 1, packetSize));
+                    this.AppendToLog(new ReadOnlySpan<byte>(bytes, pos - 1, packetSize));
                     pos += packetSize - 1;
                 }
             }
         }
 
+        void AppendToLog(ReadOnlySpan<byte> span)
+        {
+            this.log.Enqueue(span);
+            if (this.traceLogDetails)
+            {
+                this.TraceLogDetail("Appended", this.log.TailAddress, span);
+            }
+        }
+
+        void TraceLogDetail(string operation, long nextCommitLogPosition, ReadOnlySpan<byte> span)
+        {
+            byte firstByte = span[0];
+            char f = ((firstByte & first) != 0) ? 'F' : '.';
+            char l = ((firstByte & last) != 0) ? 'L' : '.';
+            byte[] crc = Crc32.Hash(span);
+            this.traceHelper.FasterStorageProgress($"CommitLogEntry {operation} {nextCommitLogPosition} {f}{l} length={span.Length} crc={crc[0]:X2}{crc[1]:X2}{crc[2]:X2}{crc[3]:X2}");
```

Comment:
I assume this CRC is related to error correction codes. Why do we need to log this and how can it help us debug what went wrong?

Reply:
In particular, also use some clarity this `first` and `last` comparisons. Not terribly familiar with this byte checks.

Reply:
I am just using the CRC as an easy way to do checksum. Means I can check whether FASTER is reading back the same bytes as it wrote.

first and last are used by the fragmentation/reassembly logic. When writing a packet that is too large for FASTER it gets split and the first packet has a FIRST flag, the last packed has a LAST flag. If there is only one packet it has both a FIRST and LAST flag.

Reply:
Thanks. As a nit: can you please write this information down as a comment in this method? I think it's useful context